src/osd/OSD.cc

```diff
@@ -9866,6 +9867,19 @@ bool OSD::maybe_override_options_for_qos()
   return false;
 }
 
+void OSD::maybe_override_cost_for_qos()
+{
+  // If the scheduler enabled is mclock, override the default PG deletion cost
+  // so that mclock can meet the QoS goals.
+  if (cct->_conf.get_val<std::string>("osd_op_queue") == "mclock_scheduler" &&
+      !unsupported_objstore_for_qos()) {
+    dout(1) << __func__
+            << ": Changing PG deletion cost for QoS" << dendl;
+    uint32_t pg_delete_cost = 15728640;
```

Comment:
Change to uint64_t as 'size' spec in the yaml file translates to uint64_t. See /src/common/options.h.

Reply:
Ah thanks for that!